app/lib/features/pins/providers/notifiers/pins_notifiers.dart

```diff
@@ -0,0 +1,144 @@
+    try {
+      return await client.pin(arg);
+    } catch (e) {
+      return await client.waitForPin(arg, null);
+    }
+    // this might throw internally
+  }
+
+  @override
+  Future<ActerPin> build(String arg) async {
+    final client = ref.watch(alwaysClientProvider);
+    _listener = client.subscribeStream(arg); // stay up to date
+    _listener.forEach((e) async {
+      state = await AsyncValue.guard(() => _getPin());
+    });
+    return _getPin();
+  }
+}
+
+class AsyncSpacePinsNotifier
+    extends AutoDisposeFamilyAsyncNotifier<List<ActerPin>, Space> {
+  late Stream<void> _listener;
+
+  Future<List<ActerPin>> _getPins() async {
+    return (await arg.pins()).toList(); // this might throw internally
+  }
+
+  @override
+  Future<List<ActerPin>> build(Space arg) async {
+    final client = ref.watch(alwaysClientProvider);
+    final spaceId = arg.getRoomId();
+    _listener = client.subscribeStream('$spaceId::PINS'); // stay up to date
+    _listener.forEach((e) async {
+      state = await AsyncValue.guard(() => _getPins());
+    });
+    return _getPins();
+  }
+}
+
+class PinEditNotifier extends StateNotifier<PinEditState> {
+  final ActerPin pin;
+  final Ref ref;
+  PinEditNotifier({required this.pin, required this.ref})
+      : super(
+          const PinEditState(
+            title: '',
+            link: '',
+            editMode: false,
+          ),
+        ) {
+    _init();
+  }
+
+  void _init() {
+    final msgContent = pin.contentText();
+    String plainText = '';
+    String? formattedBody;
+    if (msgContent != null) {
+      if (msgContent.formattedBody() != null) {
+        formattedBody = msgContent.formattedBody();
+      } else {
+        plainText = msgContent.body();
+      }
+    }
+    state = state.copyWith(
+      title: pin.title(),
+      link: pin.isLink() ? pin.url() ?? '' : '',
+      plain: plainText,
+      htmlBody: formattedBody,
+    );
+  }
+
+  void setTitle(String title) => state = state.copyWith(title: title);
+
+  void setLink(String link) => state = state.copyWith(link: link);
+
+  void setPlainText(String text) => state = state.copyWith(plain: text);
+
+  void setHtml(String? html) => state = state.copyWith(htmlBody: html);
+
+  void setEditMode(bool editMode) => state = state.copyWith(editMode: editMode);
+
+  Future<String> onSave(BuildContext context) async {
+    EasyLoading.show(status: 'Updating Pin');
+    try {
+      final updateBuilder = pin.updateBuilder();
+      updateBuilder.title(state.title);
+      if (pin.isLink()) {
+        updateBuilder.url(state.link);
+      }
+      updateBuilder.contentText(state.plain);
+      if (state.htmlBody != null) {
+        updateBuilder.contentMarkdown(state.htmlBody!);
+      }
+      await updateBuilder.send();
+      final updatedPin = await pin.refresh();
+      // make sure we refresh the pin over list too.
+      ref.invalidate(pinsProvider);
+      ref.invalidate(spacePinsProvider);
+      ref.invalidate(pinsFoundProvider);
```

Comment:
```suggestion
```
This shouldn't be necessary, but it currently is as the `pins`-listening key was wrong...